File: BocetoV2/Clases/Tarea.java
import java.time.LocalDateTime;
import java.util.Objects;
import java.util.UUID;

public class Tarea {
    private String id;
    private String titulo;
    private String descripcion;
    private LocalDateTime fecha;
    private EtiquetaTarea etiqueta;
    private boolean completada;

    public Tarea() {}
    public Tarea(String titulo, String descripcion, LocalDateTime fecha, EtiquetaTarea etiqueta) {
        this.id = UUID.randomUUID().toString();
        this.titulo = titulo;
        this.descripcion = descripcion;
        this.fecha = fecha;
        this.etiqueta = etiqueta;
        this.completada = false;
    }

    public String getId() { return id; }
    public void setId(String id) { this.id = id; }
    public String getTitulo() { return titulo; }
    public void setTitulo(String titulo) { this.titulo = titulo; }
    public String getDescripcion() { return descripcion; }
    public void setDescripcion(String descripcion) { this.descripcion = descripcion; }
    public LocalDateTime getFecha() { return fecha; }
    public void setFecha(LocalDateTime fecha) { this.fecha = fecha; }
    public EtiquetaTarea getEtiqueta() { return etiqueta; }
    public void setEtiqueta(EtiquetaTarea etiqueta) { this.etiqueta = etiqueta; }
    public boolean isCompletada() { return completada; }
    public void setCompletada(boolean completada) { this.completada = completada; }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Tarea tarea = (Tarea) o;
        return Objects.equals(id, tarea.id);
    }

    @Override
    public int hashCode() { return Objects.hash(id); }

    @Override
    public String toString() {
        return "Tarea{" + "titulo='" + titulo + '\'' + ", completada=" + completada + '}';
    }
}

File: App/AppV1/app/src/main/java/com/pmdm/appV1/models/Tarea.kt
package com.pmdm.appV1.models

import java.time.LocalDateTime
import java.util.UUID

// Asumo que EtiquetaTarea es un Enum que vendrá después, lo importaremos aquí.
// Si no existe, puedes crearlo en la carpeta 'enums'.

data class Tarea(
    var id: String = UUID.randomUUID().toString(),
    var titulo: String = "",
    var descripcion: String = "",
    var fecha: LocalDateTime = LocalDateTime.now(), // Valor por defecto seguro
    var etiqueta: EtiquetaTarea? = null,
    var completada: Boolean = false
) {
    // Kotlin genera automáticamente equals(), hashCode(), toString() y getters/setters.
}
